cpptests/t_basic.cpp

```diff
@@ -2,11 +2,32 @@
 #include <string>
 #include <vector>
 #include <cstdarg>
+#include <cstdio>
+#include <climits>          // LLONG_MAX
+
 #include "hiredis.h"
 
 class FormatterTest : public ::testing::Test {
 };
 
+class ReplyReaderTest : public ::testing::Test {
+protected:
+    redisReader *reader;
+    void *reply = nullptr;;
+    int ret;
+    int i;    
+    
+    virtual void SetUp() {
+        reader = redisReaderCreate();
+    }
+    virtual void TearDown() {
+        if(reply == nullptr) { freeReplyObject(reply); }
```

Comment:
!= nullptr...

Or maybe better yet, provide a C++ wrapper class around Reply? (not necessary, but a good exercise)